### DISCUSSION TITLE
2022.12.3.5

### DISCUSSION THREAD

**Discussion by alternateal** [tesla-android/issue-tracker#6] [Q&A] [answered]
I have got both the Pis up and running at I can browse both Android and CarPlay perfectly using my iPad connected using Safari. When I put the rig in my car, I can get it to connect to WiFi, but then when I browse on the Tesla I get about 5 seconds usage before I have to reload the web-page - see attached video link (more than 10Mb so won’t let me attach it here)

https://share.icloud.com/photos/0c0zZhOdu-XMxyWUnkIBJFdSQ

This has to be something car related, as it is only the display that is affected both for touch screen and output to the screen. 

I have also rebooted the system using both buttons on the steering wheel and that has not made any difference to what I see

**Comment by mikegapinski** [comment]
I have 2022.12.3.2, so it should be close enough to what you have...

Of the touchscreen would stop working you should see a dialog on the bottom of the page... 

1. What is your region? I know that Australia has some extra browser restrictions, might need to get around that...
2. Ryzen or Intel? If Intel, is it 2021+?(USB-A in the old center console might not deliver enough power, try 12V outlet with adapter)
3. Is the Tesla-Android Wi-Fi in the car still connected after the video breaks? 
4. Does the internet work in the car when you are connected to Tesla-Android Wi-Fi? Car can disconnect the network if the modem is not active for some time.
5. You can try an alternative video streaming mode by putting the car in drive before loading the page. After 5 seconds it should switch to MJPEG based stream and the screen should say "Low quality mode forced by loading Tesla Android in motion, reload the page while in Park". If that does not happen try to ssh into Linux Pi and do sudo systemctl stop janus and reload the page.
6. Open http://3.3.3.1:8001/stream and see if mjpeg without the flutter app works.

If number 5 or 6 moves the needle in the right direction for you I might know what is going on, something on the frontend relating to WebRTC streaming or handling it's state. This can be either related to 2022.12.3.5 breaking things a little or region restrictions in the car.

Let me know what you find out.

EDIT: after taking a closer look, if Chrome says connection interrupted this has to be network related I guess... webrtc won't throw something like this, interesting

**Reply by alternateal** [reply]
Fixed - helps if I plug the LTE modem into the correct Raspberry Pi - ie the Linux Pi not the Android Pi. All appears to be working correctly now. Thank you for this - it is a superb leap forward. Full test tomorrow when on way to work.

**Comment by alternateal** [comment, marked as answer]
The answer is to ensure the LTE modem is plugged into the Linux Pi not the Android Pi
